### PR TITLE
CircleCI jacoco test coverage

### LINKED ISSUE #135: Add code coverage reporting
We should have code coverage reporting to show/verify increasing test coverage:
- Add JaCoCo plugin to Maven
- [Upload](https://circleci.com/docs/2.0/code-coverage/#java) coverage report to CircleCI
- [Upload](https://github.com/trautonen/coveralls-maven-plugin) coverage report to [Coveralls](https://coveralls.io) (free for OSS)
- Add GitHub [check](https://github.com/marketplace/coveralls) that coverage has not decreased
- Add coverage badge to README

### PR DESCRIPTION
**Pull request by calvinchengx**
Resolves #135 partially
* Add JaCoCo plugin to Maven
* Upload coverage report to CircleCI
* Upload coverage report to Coveralls (free for OSS)

To be included in a separate, subsequent PR
* Add GitHub check that coverage has not decreased
* Add coverage badge to README

Signed-off-by: Calvin Cheng <calvin@hedera.com>

### CODE REVIEW

**Comment by calvinchengx** [comment]
> Add GitHub check that coverage has not decreased

How would we implement this, @steven-sheehy ?

Besides this, all that's left is to copy-and-paste in the coveralls coverage badge

File: .circleci/config.yml [hunk 1]
```diff
@@ -45,10 +45,15 @@ jobs:
           command: ./mvnw install
       - store_test_results:
           path: target/surefire-reports
+      - store_artifacts:
+          path: target
```
[the remark this answers is not in the document]
Comment (calvinchengx) [reply .circleci/config.yml]: Just the index.html file?

Or `target/site/jacoco` ?

[the remark this answers is not in the document]
Comment (steven-sheehy) [reply .circleci/config.yml]: index.html

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [![CircleCI](https://circleci.com/gh/hashgraph/hedera-mirror-node/tree/master.svg?style=shield)](https://circleci.com/gh/hashgraph/hedera-mirror-node/tree/master)
 ![GitHub](https://img.shields.io/github/license/hashgraph/hedera-mirror-node)
+![Coveralls github branch](https://img.shields.io/coveralls/github/hashgraph/hedera-mirror-node/master)
```
[the remark this answers is not in the document]
Comment (calvinchengx) [reply README.md]: updated.

**Pull request merged**